crates/context_aware_config/src/api/dimension/types.rs

```diff
@@ -13,60 +18,17 @@ pub struct CreateReq {
     pub change_reason: String,
 }
 
-#[derive(Debug, Deserialize, AsRef, Deref, DerefMut, Into, Clone, Default)]
-#[serde(try_from = "i32")]
-pub struct Position(i32);
-impl Position {
-    fn validate_data(position_val: i32) -> Result<Self, String> {
-        if position_val < 0 {
-            Err("Position should be greater than equal to 0".to_string())
-        } else {
-            Ok(Self(position_val))
-        }
-    }
-}
-
-impl TryFrom<i32> for Position {
-    type Error = String;
-    fn try_from(value: i32) -> Result<Self, Self::Error> {
-        Self::validate_data(value)
-    }
-}
-
-#[derive(Debug, Deserialize, Clone)]
+#[derive(Debug, Deserialize, Clone, AsChangeset)]
+#[diesel(table_name = dimensions)]
 pub struct UpdateReq {
     pub position: Option<Position>,
     pub schema: Option<Value>,
-    pub function_name: Option<FunctionNameEnum>,
+    #[serde(default, deserialize_with = "deserialize_function_name")]
```

Comment:
default is not needed, right ?
```suggestion
    #[serde(deserialize_with = "deserialize_function_name")]
```